ocaml/testsuite/tests/typing-layouts/jkinds.ml

```diff
@@ -84,6 +87,372 @@ Uncaught exception: Misc.Fatal_error
+         of the definition of b at line 2, characters 0-30.
+|}]
+
+type a : value mod global unique many uncontended portable external_
+type b : value mod local shared once contended nonportable internal = a
+[%%expect{|
+type a : value mod global unique many uncontended portable external_
+type b = a
+|}]
+
+type a : value mod global unique once uncontended portable external_
+type b : value mod local shared many uncontended nonportable internal = a
+[%%expect{|
+type a : value mod global unique once uncontended portable external_
+Line 2, characters 0-73:
+2 | type b : value mod local shared many uncontended nonportable internal = a
+    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Error: The layout of type a is immediate, because
+         of the definition of a at line 1, characters 0-68.
+       But the layout of type a must be a sublayout of value, because
+         of the definition of b at line 2, characters 0-73.
+|}]
+
+(********************************************************)
+(* Test 3: Abbreviation primitives are properly defined *)
+(* CR layouts: when we have abbreviations, these tests can become less verbose *)
+
+type a : any
+type b : any mod local shared once contended nonportable internal = a
+type c : any mod local shared once contended nonportable internal
+type d : any = c
+[%%expect{|
+type a : any
+type b = a
+type c : any mod local shared once contended nonportable internal
+type d = c
+|}]
+
+type a : value
+type b : value mod local shared once contended nonportable internal = a
+type c : value mod local shared once contended nonportable internal
+type d : value = c
+[%%expect{|
+type a : value
+type b = a
+type c : value mod local shared once contended nonportable internal
+type d = c
+|}]
+
+type a : void
+type b : void mod local shared once contended nonportable internal = a
+type c : void mod local shared once contended nonportable internal
+type d : void = c
+[%%expect{|
+Line 1, characters 9-13:
+1 | type a : void
+             ^^^^
+Error: Layout void is more experimental than allowed by the enabled layouts extension.
+       You must enable -extension layouts_alpha to use this feature.
+|}]
+
+type a : immediate
+type b : value mod global unique many uncontended portable external_ = a
+type c : value mod global unique many uncontended portable external_
+type d : immediate = c
+[%%expect{|
+type a : immediate
+type b = a
+type c : value mod global unique many uncontended portable external_
+type d = c
+|}]
+
+type a : immediate64
+type b : value mod global unique many uncontended portable external64 = a
+type c : value mod global unique many uncontended portable external64
+type d : immediate64 = c
+[%%expect{|
+type a : immediate64
+type b = a
+type c : value mod global unique many uncontended portable external64
+type d = c
+|}]
+
+type a : float64
+type b : float64 mod global unique many uncontended portable external_ = a
```

Comment:
This is a bug. That is, `float64` should really just be the layout, not with any mode crossing. We've been unwilling to fix the bug until we had syntax to get the mode crossing separately, and now we do! But probably best to fix in a future PR. I've made ticket 3157.